addon/components/column-chart/component.js

```diff
@@ -411,11 +412,15 @@ export default BaseChartComponent.extend({
 
         // Choose the tallest bar in the stack (lowest y value) and place the data labels above that.
         // Avoids label falling under any bar in the stack.
-        let yValues = [];
+        let yValues = { y: [], height: [] };
         this.get('chart').selectAll('.sub rect.bar').each(function () {
-            yValues.push(parseInt(d3.select(this).attr('y')));
+            yValues.y.push(parseInt(d3.select(this).attr('y')));
+            yValues.height.push(parseInt(d3.select(this).attr('height')));
         });
-        const maxLabelY = Math.min(...yValues);
+        const maxLabelY = Math.min(...yValues.y);
+        const maxLabelYHeight = Math.max(...yValues.height);
+        // to indicate if chart has all negative values, so that data labels could be rendered at negative side of x -axis.
+        const allNegativeValues = this.get('yAxis').allNegative;
```

Comment:
Is `allNegative` a new property that consumers will pass in on `yAxis`? If so, we can determine if all values are negative inside this component rather than require consumers to pass in the property.

Reply:
It's calculated as a computed property inside the controller of survey performance detail and is assigned to the object of yAxis.

Reply:
added a new property, bottomLabelPosition= true | false